src/cmake/feature-flag.cmake

```diff
@@ -0,0 +1,33 @@
+# Copyright (C) Canonical, Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+include(CMakeDependentOption)
+
+# FIXME: How should we force all the various feature flags to change their
+# cached values if we change this setting after the first configure?
+option(MP_DISABLE_ALL_FEATURES "disable all optional features" OFF)
```

Comment:
iirc we were going to lay this out with the logic inverted to what it is here. i.e. `MP_ENABLE_ALL_FEATURES` with a default value of `ON`.

Reply:
Based on discussion in the standup, this is now `MP_ALLOW_OPTIONAL_FEATURES`.